platform/darwin/src/NSExpression+MGLAdditions.mm

```diff
@@ -878,9 +878,25 @@ + (instancetype)expressionWithMGLJSONObject:(id)object {
                 NSExpression *expression = [NSExpression expressionWithMGLJSONObject:argumentObjects[index]];
                 NSMutableDictionary *attrs = [NSMutableDictionary dictionary];
                 if ((index + 1) < argumentObjects.count) {
-                    attrs = argumentObjects[index + 1];
+                    attrs = [NSMutableDictionary dictionaryWithDictionary:argumentObjects[index + 1]];
                 }
                 
+                if (attrs.count) {
+                    if (attrs[MGLFontNamesAttribute]) {
+                        NSArray *fontNames = (NSArray *)attrs[MGLFontNamesAttribute];
+                        attrs[MGLFontNamesAttribute] = fontNames[1];
+                    }
+                    if (attrs[MGLFontColorAttribute] && [attrs[MGLFontColorAttribute] isKindOfClass:[NSArray class]]) {
+                        NSArray *colorArray = attrs[MGLFontColorAttribute];
```

Comment:
Minor FYI: we have a `MGL_OBJC_DYNAMIC_CAST` macro that can help with readability
for code like this.